packages/loader/container-loader/src/prefetchDocumentStorageService.ts

```diff
@@ -27,27 +27,24 @@ export class PrefetchDocumentStorageService extends DocumentStorageServiceProxy
         return p;
     }
 
-    public async read(blobId: string): Promise<string> {
-        return this.cachedRead(blobId);
-    }
-
     public stopPrefetch() {
         this.prefetchEnabled = false;
         this.prefetchCache.clear();
     }
 
     // eslint-disable-next-line @typescript-eslint/promise-function-async
-    private cachedRead(blobId: string): Promise<string> {
+    private async cachedRead(blobId: string): Promise<string> {
+        const blob = await this.internalStorageService.readBlob(blobId);
         if (this.prefetchEnabled) {
             const prefetchedBlobP: Promise<string> | undefined = this.prefetchCache.get(blobId);
             if (prefetchedBlobP !== undefined) {
                 return prefetchedBlobP;
             }
-            const prefetchedBlobPFromStorage = this.internalStorageService.read(blobId);
+            const prefetchedBlobPFromStorage = blobToString(blob);
```

Comment:
Adapters like that should do no conversions. They should read binary blob, cache binary blob and return binary blob.
Otherwise they risk converting real binary data to string and explode.